### PR TITLE
Pull package readmes during build time

### PR DESCRIPTION
**Pull request by spejamchr**
~https://github.com/execonline-inc/CooperTS/blob/907663b002021e0669bbf8cf701ae98c32201dee/packages/numbers/README.md~

Instead of manually copying package readmes into the docs (and then maintaining them), pull them in during the build stage.

### CODE REVIEW

**Review by oryolo — COMMENTED** [review]
Looks good 💯  Maybe copy this file also to `/guide`?

File: packages/numbers/README.md [hunk 1]
```diff
@@ -1,11 +1,59 @@
...
+### `parseIntT`
+
+Attempts to parse an integer from a string, using `Task<NumberParseFailure, number>` to wrap the
+success or failure of the parse.
+
+```ts
+import { parseIntT } from '@execonline-inc/numbers';
+
+parseIntT('123').getOrElseValue(0) // returns `123`
+parseIntT('123.42').getOrElseValue(0) // returns `123`
```
[the remark this answers is not in the document]
Comment (spejamchr) [reply packages/numbers/README.md:45]: Ah, oops, I messed up my copy-pasting. Thanks!

**Comment by pjrebsch** [comment]
I'd like to see if we can explore de-duplication now to avoid setting a precedent for writing more duplicated documentation in the meantime.

**Comment by spejamchr** [comment]
He @pjrebsch I got this working. During build time it fetches the readmes and package.json files for all the packages and uses them to build out a `packages/{slug}` page for each one.

File: docs/utils/getPackages.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  glob.sync(localPackageDirFromSlug('*')) as PackageDir[];
+
+const slugFromPackageDir = (packageDir: PackageDir): string => path.basename(packageDir);
+
+const getLocalPackageSlugs = (): string[] => getLocalPackageDirs().map(slugFromPackageDir);
+
+const localPackageMetadataFromSlug = (slug: string): PackageMetadata =>
+  requirePackageMetadataDuringBuild(
+    JSON.parse(
+      fs.readFileSync(`${localPackageDirFromSlug(slug)}/package.json`, { encoding: 'utf-8' })
```
Comment (pjrebsch) [inline docs/utils/getPackages.ts:41]: Filesystem interaction should probably be a Task. See `exec_online/.github/actions/lib/packages/filesystem` for some existing implementations which may become another CooperTS package in the future.

File: docs/utils/getPackages.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+export const getPackageDataFromSomeSlug = (slug: string): Task<HttpError, PackageData> => {
+  if (Object.hasOwn(festivePossumReadmes, slug)) {
+    return festivePossumPackageData(
+      festivePossumPackage(slug, festivePossumReadmes[slug as keyof typeof festivePossumReadmes])
+    );
+  } else {
+    return Task.succeed<HttpError, PackageData>(localPackageDataFromSlug(slug));
+  }
+};
```
Comment (pjrebsch) [inline docs/utils/getPackages.ts:123]: To remove type assertion from this code,

```ts
const festivePossumReadmes = {
// ...
} as const;

type FestivePossumPackageName = keyof typeof festivePossumReadmes;
```

```ts
const isFestivePossumPackageName = (slug: string): slug is FestivePossumPackageName =>
  Object.hasOwn(festivePossumReadmes, slug);

export const getPackageDataFromSomeSlug = (slug: string): Task<HttpError, PackageData> => {
  if (isFestivePossumPackageName(slug)) {
    return festivePossumPackageData(festivePossumPackage(slug, festivePossumReadmes[slug]));
  } else {
    return Task.succeed<HttpError, PackageData>(localPackageDataFromSlug(slug));
  }
};
```

File: docs/pages/packages.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+const Packages: React.FC<Props> = ({ allPackageData }) => {
+  return (
+    <ul className={styles.bulletList}>
+      {allPackageData
+        .sort((a, b) => (a.metadata.name > b.metadata.name ? 1 : -1))
+        .map(({ slug, metadata: { name, description } }) => (
+          <React.Fragment key={slug}>
+            <li>
```
[the remark this answers is not in the document]
Comment (spejamchr) [reply docs/pages/packages.tsx:17]: Before I was rendering a <dl>, so the fragment had both <dt> and <dd>. I'll change it.

File: docs/utils/getPackages.ts (hunk 3 above)
Comment (spejamchr) [reply docs/utils/getPackages.ts:123]: I ended up making `festivePossumReadmes` a `Map` type instead. It was a bit easier to work with.

File: docs/GetPackages/index.ts [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+  Task.succeed<HttpError, { slug: string }>({ slug: paths.slug })
+    .assign('metadata', festivePossumPackageMetadata(paths))
+    .assign('readme', festivePossumPackageReadmeContent(paths));
+
+type GetFestivePossumPackagesError = HttpError;
+
+const getAllFestivePossumPackageData = (): Task<
+  GetFestivePossumPackagesError,
+  Array<PackageData>
+> => Task.all(festivePossumPaths().map(festivePossumPackageData));
```
Comment (spejamchr) [inline docs/GetPackages/index.ts:109]: I updated this to use `Task.all`, which runs the tasks in parallel, and the dev experience is much better. Before, the page took several seconds to load while fetching all the packages. Now it takes about 1s for the first page load.

**Comment by spejamchr** [comment]
@pjrebsch This is ready for another round.

**Review by kofno — APPROVED** [review]

**Pull request merged**